Sources/SwiftFusion/Nonlinear/NonlinearFactorGraph.swift

```diff
@@ -0,0 +1,114 @@
+import TensorFlow
+
+public struct NonlinearFactorGraph {
+  var factor1s: [Factor1] = []
+  var factor2s: [Factor2] = []
```

Comment:
Not polymorphic over the variable types yet. Not even polymorphic over the arities of the factors.

Reply:
I think the variable types are only defined/needed in the factors - not here.